tests/test_adaptive_support.py

```diff
@@ -0,0 +1,133 @@
+import logging
+import os
+from unittest import TestCase
+
+import numpy as np
+import pytest
+
+from aspire.denoising import adaptive_support
+from aspire.image import Image
+from aspire.source import ArrayImageSource
+from aspire.utils import circ, gaussian_2d, hankel
+
+logger = logging.getLogger(__name__)
+
+DATA_DIR = os.path.join(os.path.dirname(__file__), "saved_test_data")
+
+
+class AdaptiveSupportTest(TestCase):
+    def setUp(self):
+
+        self.resolution = resolution = 1025
+        self.sigma = sigma = 128
+        self.n_disc = n_disc = 10
+
+        # Create 2D Gaussian as initial array.
+        discs = np.tile(
+            gaussian_2d(resolution, sigma_x=sigma, sigma_y=sigma),
+            (n_disc, 1, 1),
+        )
+
+        # Add varying radius of solid disc.
+        #  The Fourier transform will yield Airy disc
+        #  which has more interesting content in F space.
+        for d in range(n_disc):
+            discs[d] = discs[d] + circ(resolution, radius=(d + 1) ** 2)
+
+        self.img_src = ArrayImageSource(Image(discs))
```

Comment:
This is only used in one of the tests. Might be more straightforward to move it to that test.

Reply:
used in three tests ,  two test functions... but I see what you mean.. can replace some with dummy data.  Moved into the child functions.

Reply:
Right I missed the bad input tests. Thanks!